### ISSUE TITLE
[clang-tidy] false-positive bugprone-branch-clone when switch case fallthrough with some Side Effects exp

### ISSUE THREAD

**Issue by hstk30-hw** [llvm/llvm-project#87146]
https://godbolt.org/z/zshW85bME

```
int test1(int a) {
    int b = 0;
    switch (a) {
        case 1: b++;  // fall through
        case 2: b++;  // fall through
        case 3: b++;  // fall through
        case 4: b++;  // fall through
        default: ;
    }
    return b;
}
```

In the case, every `case` is fall through. 
So, in fact, the branchs is not identical in semantic.

**Comment by 5chmidti** [comment]
You can provide the `[[fallthrough]]` label to mark the case as such, and the check will no longer diagnose those cases: https://godbolt.org/z/a1zb5GYoP 

```c++
int test1(int a) {
    int b = 0;
    switch (a) {
        case 1: b++; [[fallthrough]];
        case 2: b++; [[fallthrough]];
        case 3: b++; [[fallthrough]];
        case 4: b++; [[fallthrough]];
        default: ;
    }
    return b;
}
```
However, given that you provided an example with C selected, that would require C23. You could use the `__attribute__((fallthrough))` syntax instead if you are constrained to an earlier version of C:
```c++
int test1(int a) {
    int b = 0;
    switch (a) {
        case 1:
            b++; __attribute__((fallthrough));
        case 2:
            b++; __attribute__((fallthrough));
        case 3:
            b++; __attribute__((fallthrough));
        case 4:
            b++; __attribute__((fallthrough));
        default:;
    }
    return b;
}
```
https://godbolt.org/z/7cPzW1r4Y

**Comment by hstk30-hw** [comment]
Thx you reply!

**Comment by hstk30-hw** [comment]
Same as https://github.com/llvm/llvm-project/issues/47588
Fixed by  https://reviews.llvm.org/D147889
